File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.c
int chalkReplacer(int* chalk, int chalkSize, int k) {
    long long sum=0;
    for(int i=0; i<chalkSize; ++i) {
        sum+=chalk[i];
    }
    k=k%sum;
    int j=0;
    while(k>0) {
        if (k >= chalk[j]) {
            k-= chalk[j];
        } else {
            return j;
        }
        j=(j+1)%chalkSize;
    }
    return j;
}

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        i = 0
        n = len(chalk)
        k = k%sum(chalk)
        while k > 0:
            if k >= chalk[i]:
                k-=chalk[i]
            else:
                return i
            i=(i+1)%n
        return i
            

        